pm/README.md

```diff
@@ -0,0 +1,37 @@
+# Probabilistic Micropayments (PM) Client
+
+The `pm` package contains a client implementation of the [Livepeer PM protocol](https://github.com/livepeer/wiki/blob/master/spec/streamflow/pm.md). The main areas of interest are:
+
+- A `Ticket` type which defines the ticket format that both senders and recipients accept.
+- A `Sender` type which is responsible for creating tickets.
+- A `Recipient` type which is responsible for validating/receiving tickets and redeeming winning tickets using the `TicketBroker` interface (implemented by an Ethereum smart contract). This type will be able to translate a `Ticket` type into the correct format expected by the `TicketBroker`
+
+In practice, the package user will need to define a communication channel (i.e. HTTP connection) and a network message format (i.e. protocol buffer) that peers will use to send and receive tickets. The sending peer will use the `Sender` type and the receiving peer will use the `Recipient` type.
+
+## High Level Flow
+
+![flow diagram](diagram.png)
+
+[Sequence Diagram Source](https://sequencediagram.org/index.html#initialData=C4S2BsFMAIGVIHYBMQIObQGTQEqQMaQgBuqGAKiPgNaTADOAUI3viAA4iLAC0AfPGSQATgC5ooGnQAKAQ2GyAtk0YIA9sBhriIuIiQjxsYPODx69EGoQAKSbWByFygJTN1m6Nt2CDYiVQO9NAAvNAAwsKQspqUUsAAQjH4ABY2CG6Mvrr8uAQcXAjA4ggB8SoeWjrCeWyc3OKsRDpxDnaBdC7QAGZqNbLg4NCl9nQVGlW6rAUNeQaQigDqqAhkrXTt8V29-YPQAO4rZGVB7hNe1XpC-qMMoRFRMZDriclpGczZNbnT9UUlJzGZ083hqv0KxVqzWeHWAmwc2z60AGQxGsPGIMu4NmeHmSyO6Be8M6PSRKIOBIwt3oQA)
+
+## RNG
+
+A `Sender` and a `Recipient` execute a collaborative commit-reveal protocol in order to generate the random number used to determine if a ticket won or not. The random number construction is:
+
+```
+keccak256(abi.encodePacked(senderSig, recipientRand))
+```
+
+The `abi.encodePacked()` Solidity built-in is used to match the behavior of the contract implementing the `TicketBroker` interface. `senderSig` is a signature produced by `Sender` over the hash of the ticket and `recipientRand` is the pre-image of the `recipientRandHash` that is included in the ticket.
+
+Before a `Sender` can start creating and sending tickets, it must first fetch ticket parameters from a `Recipient`. When generating ticket parameters, a `Recipient` will generate a random number `recipientRand`, compute the hash of `recipientRand` as `recipientRandHash` and include the hash with the ticket parameters. Once a `Sender` has a set of ticket parameters with a `recipientRandHash`, the `Sender` will use a monontonically increasing `senderNonce` with each new ticket that it creates.
```

Comment:
It might be a good idea to provide some extra context around `TicketParams` here , perhaps we can use the commented protobuf definitions to provide more clarity here?